ios/FluentUI/Vnext/HUD/HeadsUpDisplay.swift

```diff
@@ -0,0 +1,227 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+import UIKit
+
+/// Defines the content type of the Heads-up display.
+/// The `.activity` value will make the HUD display an Activity Indicator.
+/// The `.success` and `.failure` values will make the HUD display its default images for the cases.
+/// The `.custom` value allows a UIImage to be passed as a parameter for the HUD to display.
+public enum HUDType: Equatable, Hashable {
+    case activity
+    case custom(image: UIImage)
+    case failure
+    case success
+
+    public func hash(into hasher: inout Hasher) {
+        switch self {
+        case .custom(let image):
+            hasher.combine(image)
+        case .activity, .success, .failure:
+                break
+        }
+    }
+}
+
+/// Properties available to customize the state of the Heads-up Display state.
+public protocol MSFHUDState {
+
+    /// Label string presented by the Heads-up Display.
+    var label: String? { get set }
+
+    /// The action executed when the Heads up display is tapped.
+    var tapAction: (() -> Void)? { get set }
+
+    /// The `HUDType` enum value of the Heads-up Display.
+    var type: HUDType { get set }
+
+    /// Design token set for this control, to use in place of the control's default Fluent tokens.
+    var overrideTokens: HeadsUpDisplayTokens? { get set }
+}
+
+/// View that represents the Heads-up Display.
+public struct HeadsUpDisplay: View, ConfigurableTokenizedControl {
```

Comment:
I'm debating about how I feel re: this name. I know that this is what HUD stands for, but I don't think I've ever seen it spelled out before in a UIKit/AppKit context. Out of curiosity, was there a reason not to just call it `HUD`?

Reply:
The main class in UIKit is called HUD. I debated about naming it FluentHUD (to follow the Button pattern), but since this is not a clash with SwiftUI types, "HeadsUpDisplay" was the best one I thought about.